### ISSUE TITLE
Setting colour in slicer with single colour print

### ISSUE THREAD

**Issue by rkarlsba** [bambulab/BambuStudio#673]
Hi all

If I set the colour to number 2 in the slicer and press print, I still need to set it again in the filament selector when the print popup comes. Is it possible to remove this last step if I've set the colour in the slicer in the first place?

roy

**Comment by MadMax389** [comment]
I believe if you match the color in the slicer to the color selected in the AMS device screen (regardless of the material selected) it will select the right AMS slot when sending the print. I will set up the colors is the AMS device screen first, then sync the AMS colors in the slicer. Then you can select a different material in the slicer and still maintain the proper color. When you send the print it selects the right AMS slot.
![image](https://user-images.githubusercontent.com/96731892/202901047-bfb7ecff-778a-4650-83be-f31b205ccb39.png)
![image](https://user-images.githubusercontent.com/96731892/202901207-4dfc3cfb-d1f9-4af7-aad7-81123ad42ef6.png)

**Comment by rkarlsba** [comment]
The problem is that this doesn't work with less than four colours, as far as I can see. I just sent a two-colour job to the printer and chose the correct colours. The print screen shows up and it automatically selects filament 1 and 2, regardless of what I chose in the prepare screen. This is the problem. 

Vennlig hilsen 

roy 
-- 
Roy Sigurd Karlsbakk 
(+47) 98013356 
-- 
I all pedagogikk er det essensielt at pensum presenteres intelligibelt. Det er et elementært imperativ for alle pedagoger å unngå eksessiv anvendelse av idiomer med xenotyp etymologi. I de fleste tilfeller eksisterer adekvate og relevante synonymer på norsk.

**Comment by MadMax389** [comment]
Yes I see your problem now. I loaded two spools in slots 3 and 4 and assigned colors on the device screen (slots 1 and 2 empty). When you sync the AMS on the prepare screen it assigns the colors to slots 1 and 2 (instead of 3 and 4). Seems like a bug.

**Comment by rkarlsba** [comment]
Correct. Hope you'll sort it out :) 

Vennlig hilsen 

roy 
-- 
Roy Sigurd Karlsbakk 
(+47) 98013356 
-- 
I all pedagogikk er det essensielt at pensum presenteres intelligibelt. Det er et elementært imperativ for alle pedagoger å unngå eksessiv anvendelse av idiomer med xenotyp etymologi. I de fleste tilfeller eksisterer adekvate og relevante synonymer på norsk.

**Comment by MadMax389** [comment]
LOL. Not me, I'm just another user ;)
